assets/src/edit-story/app/media/utils/useProcessVideo.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+/**
+ * Internal dependencies
+ */
+import { useStory, useAPI } from '../../';
+import getFileFromUrl from './getFileFromUrl';
+
+function useProcessVideo({ uploadMedia, uploadVideoPoster }) {
+  const { updateElementsByResourceId } = useStory((state) => ({
+    updateElementsByResourceId: state.actions.updateElementsByResourceId,
+  }));
+
+  const {
+    actions: { updateMedia },
+  } = useAPI();
+
+  const updateElement = useCallback(
+    ({ oldResource, resource }) => {
+      const { id: oldId, alt, title } = oldResource;
+      updateElementsByResourceId({
+        id: oldId,
+        properties: () => {
+          return {
+            type: resource.type,
+            resource: {
+              ...resource,
+              alt,
+              title,
+            },
+          };
+        },
+      });
+    },
+    [updateElementsByResourceId]
+  );
+
+  const updateOldVideo = useCallback(
+    async (oldId, newId) => {
+      await updateMedia(oldId, {
+        meta: {
+          web_stories_transcoded_id: newId,
+        },
+      });
+    },
+    [updateMedia]
+  );
+
+  const regenerateVideo = useCallback(
+    ({ resource: oldResource }) => {
+      const { src: url, mimeType } = oldResource;
+      const onUploadSuccess = ({ resource }) => {
+        updateElement({ oldResource, resource });
+        updateOldVideo(oldResource.id, resource.id);
+
+        if (resource.type === 'video' && !resource.local) {
+          uploadVideoPoster(resource.id, resource.src);
+        }
+      };
+
+      const process = async () => {
+        const file = await getFileFromUrl(url, mimeType);
```

Comment:
Hmm doesn't look like there's a way around doing this `fetch` here, even though it seems wasteful :/

Perhaps worth testing to see whether using an array buffer is faster than a blob, but then again we can't really [stream it directly](https://web.dev/fetch-upload-streaming/) to  the server yet anyway.

Reply:
I was wonder we could reuse the [fetchFile](https://github.com/ffmpegwasm/ffmpeg.wasm/blob/9b8a5f7723dd38143602a259828c3d95eb49bcd0/src/node/fetchFile.js) in ffmpeg.wasm here. It basically do the same thing. 
I do agree that is wasteful, but I don't see a way around it. Some of these files may 100mb or larger and we do not know if the file is in the browsers cache. This seemed the safest way.

Reply:
Checking the [docs](https://developer.mozilla.org/en-US/docs/Web/API/Request/cache), it seems like if file is cached in the browser then it will download as the default behaviour. We could maybe use `force-cache` param, not sure it is needed.

Reply:
Instead of

```js
const response = await fetch(url);
const data = await response.blob();
return new File([data], name, {
type: mimeType,
});
```

in `getFileFromUrl`

we could probably just do

```js
const response = await fetch(url);
const data = await response.arrayBuffer();
return new Blob(new Uint8Array(data), { type: mimeType });
```

Reply:
file is passed to [uploadFile](https://github.com/google/web-stories-wp/blob/014162c03ad743a47db224b9e3eaafe04397ca23/assets/src/edit-story/app/uploader/useUploader.js#L138-L158). This expects a file object and not blob here. It would be a much large refactor to change this. 

What is the benefit of using the blob here?

Reply:
It'd be the more appropriate data format. Blobs can be used in many places where `File` is expected, so this should work.

Reply:
I don't understand why this would be wanted here? `uploadMedia` expects an array of file objects not blobs. To change this would be large and pointless refactor. Can you explain the benefit of the blob over a file?